shell/common/api/electron_api_native_image.h

```diff
@@ -84,15 +85,16 @@ class NativeImage : public gin_helper::Wrappable<NativeImage> {
   v8::Local<v8::Value> ToPNG(gin::Arguments* args);
   v8::Local<v8::Value> ToJPEG(v8::Isolate* isolate, int quality);
   v8::Local<v8::Value> ToBitmap(gin::Arguments* args);
+  std::vector<float> GetScaleFactors();
   v8::Local<v8::Value> GetBitmap(gin::Arguments* args);
   v8::Local<v8::Value> GetNativeHandle(gin_helper::ErrorThrower thrower);
-  gin::Handle<NativeImage> Resize(v8::Isolate* isolate,
+  gin::Handle<NativeImage> Resize(gin::Arguments* args,
                                   base::DictionaryValue options);
   gin::Handle<NativeImage> Crop(v8::Isolate* isolate, const gfx::Rect& rect);
   std::string ToDataURL(gin::Arguments* args);
   bool IsEmpty();
-  gfx::Size GetSize();
-  float GetAspectRatio();
+  gfx::Size GetSize(const base::Optional<float> scale_factor);
+  float GetAspectRatio(float scale_factor);
```

Comment:
These functions are exposed to JS, is it your intention to add extra parameters without documenting it?

In particular `getAspectRatio` is changed in a breaking way here.

I think it might be better to move the bodies of both of these functions out to helper methods are called both from the JS-exposed functions and from other methods that need to pass in the scale factor. i.e. the `GetSize` and `GetAspectRatio` functions that are exposed to JS should continue to take 0 parameters, but internally they call `GetSizeImpl(1.0)` or something.

Reply:
i did intend to do that, but i'm happy to do something else